patches/tModLoader/Terraria/ModLoader/ModAVFX.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Terraria.Graphics.Capture;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// ModAVFX is an abstract class that your classes can derive from. It serves as a container for handling exclusive AVFX content such as backgrounds, music, and water styling. 
+	/// </summary>
+	public abstract partial class ModAVFX : ModType
+	{
+		public int Type { get; internal set; }
+
+		// AVFX properties
+		public virtual ModWaterStyle WaterStyle => null;
+		public virtual ModSurfaceBgStyle SurfaceBackgroundStyle => null;
+		public virtual ModUgBgStyle UndergroundBackgroundStyle => null;
+		public virtual int Music => -1;
+		public virtual AVFXPriority Priority => AVFXPriority.None;
+
+		// Capture Camera property
+		public virtual CaptureBiome.TileColorStyle TileColorStyle => CaptureBiome.TileColorStyle.Normal;
+
+		// Methods
+		protected override void Register() {
+			Type = Loaders.AVFXs.Register(this);
+		}
+
+		/// <summary>
+		/// Forcefully registers the provided ModAVFX to Loaders.
+		/// ModBiome and direct implementations call this.
+		/// Does NOT cache the return type
+		/// </summary>
+		internal void RegisterAVFX(ModAVFX modAVFX) {
+			Loaders.AVFXs.Register(this);
+		}
+
+		/// <summary>
+		/// Returns the strength to which the AVFXEffects are active. 0% < val < 200%. Defaults to 100% (ie 100) 
+		/// </summary>
+		public virtual byte GetWeight(Player player) => 100;
```

Comment:
I've added much more documentation to hopefully make it clear what it does.
Not sure why would want to return int, when GetCorrWeight is going to bound it anyways between 0 and 200, so any additional bytes are waste.

Reply:
`int` is the preferred way to express 'this is just a number'. `byte` should only be used when the value is being stored or transmitted, in a place where the memory cost matters. It provides more to think about, is slightly harder to refactor, and makes general math harder (eg averaging). 

Similarly, it's accepted as best practice to use `int` in for loops or array indexers instead of `uint` even though the value cannot go below 0.